safenode/src/domain/dbc_genesis.rs

```diff
@@ -43,25 +54,156 @@ pub enum Error {
     FailedToParseReason(#[from] DbcError),
 }
 
-/// Create the genesis DBC.
-/// The genesis DBC is the first DBC in the network. It is created without
-/// a source transaction, as there was nothing before it.
-#[allow(clippy::result_large_err)]
-pub fn create_genesis() -> GenesisResult<Dbc> {
-    let secret_key = bls::SecretKey::from_hex(GENESIS_DBC_SK).map_err(sn_dbc::Error::Blsttc)?;
+lazy_static! {
+    /// Load the genesis DBC.
+    /// The genesis DBC is the first DBC in the network. It is created without
+    /// a source transaction, as there was nothing before it.
+    pub static ref GENESIS_DBC: Dbc = match sn_dbc::Dbc::from_hex(GENESIS_DBC_HEX) {
+        Ok(dbc) => dbc,
+        Err(err) => panic!("Failed to read genesis DBC: {err:?}"),
+    };
+}
+
+/// Return if provided DbcTransaction is genesis parent tx.
+pub(crate) fn is_genesis_parent_tx(parent_tx: &DbcTransaction) -> bool {
+    parent_tx == &GENESIS_DBC.src_tx
```

Comment:
I find it weird to have a genesis tx, with no inputs right?

Reply:
It actually has one input. But an input that doesn't exist in the network.